dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ErrorDecoder.java

```diff
@@ -117,4 +133,54 @@ private static String toString(InputStream body) throws IOException {
             return CharStreams.toString(reader);
         }
     }
+
+    private static ResponseDiagnostic diagnostic(Response response) {
+        return new ResponseDiagnostic(diagnosticArgs(response));
+    }
+
+    private static ImmutableList<Arg<?>> diagnosticArgs(Response response) {
+        ImmutableList.Builder<Arg<?>> args = ImmutableList.<Arg<?>>builder().add(SafeArg.of("status", response.code()));
+        recordHeader(HttpHeaders.SERVER, response, args);
```

Comment:
Is this safe?

Reply:
Yep, just the server name, WC uses the local service UserAgent, nginx uses `nginx`.